pyomo/common/errors.py

```diff
@@ -41,3 +41,8 @@ class InfeasibleConstraintException(PyomoException):
     the course of range reduction).
     """
     pass
+
+
+class NondifferentiableError(ValueError):
```

Comment:
This should also inherit from `PyomoException`, right?

Reply:
Yes. Good point. ```PyomoException``` did not exist when I created this PR. I'll fix it now.